src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/Source/SourceSymbolTests.vb

```diff
@@ -536,5 +536,57 @@ End Module
 
         End Sub
 
+        <WorkItem(187865, "https://devdiv.visualstudio.com:443/defaultcollection/DevDiv/_workitems/edit/187865")>
+        <Fact>
+        Public Sub DifferentMembersMetadataName()
+            Dim compilation = CompilationUtils.CreateCompilationWithMscorlib(
+<compilation>
+    <file name="a.vb">
+Delegate Sub D()
+Class C
+    Function Get_P(o As Object) As Object
+        Return o
+    End Function
+    Function p(o As Object) As Object
+        Return o
+    End Function
+    ReadOnly Property P As Object
+        Get
+            Return Nothing
+        End Get
+    End Property
+    Event E As D
+    Sub Add_E()
+    End Sub
+    ReadOnly Property add_e As Object
+        Get
+            Return Nothing
+        End Get
+    End Property
+    Declare Sub ADD_E(o As Object)
+End Class
+    </file>
+</compilation>)
```

Comment:
Are there any errors reported for this compilation? Please add AssertTheseDiagnostics call for the compilation.